### ISSUE TITLE
Shutting down lprint server results in SIGABRT

### ISSUE THREAD

**Issue by jashephe** [michaelrsweet/lprint#18]
If I start the lprint server and then exit the process with Ctrl-C, the following is printed:
```
$ lprint server -o log-file=- -o log-level=debug
I [2020-03-26T14:35:20Z] System configuration loaded, 1 printers.
I [2020-03-26T14:35:20Z] Listening for local connections at '/tmp/lprint1001.sock'.
I [2020-03-26T14:35:20Z] Starting main loop.
^CI [2020-03-26T14:35:21Z] Shutting down main loop.
lprint: thread-watch.c:171: avahi_threaded_poll_lock: Assertion `p' failed.
fish: “lprint server -o log-file=- -o…” terminated by signal SIGABRT (Abort)
```
A quick Google search would suggest that the Avahi message can be seen in other software, but I'm not sure of its significance, or why `SIGABRT` would be seen. Let me know if there's any other diagnostic information I can provide!

---

**Printer**: Dymo LabelWriter 450 Twin Turbo
**System**: `Linux <hostname> 5.5.11-1-ARCH #1 SMP Sun Mar 22 07:43:38 MDT 2020 aarch64 GNU/Linux`

**Comment by jashephe** [comment]
Note that I cannot reproduce this behavior on `Linux <hostname> 5.5.11-arch1-1 #1 SMP PREEMPT Sun, 22 Mar 2020 16:33:15 +0000 x86_64 GNU/Linux`, so it may be platform-specific.
```
$ lprint server -o log-file=- -o log-level=debug
I [2020-03-26T14:39:46Z] System configuration loaded, 1 printers.
I [2020-03-26T14:39:46Z] Listening for local connections at '/tmp/lprint1001.sock'.
I [2020-03-26T14:39:46Z] Starting main loop.
^CI [2020-03-26T14:39:47Z] Shutting down main loop.
```

**Comment by michaelrsweet** [comment]
This is likely an Avahi bug, and they like to abort() for any little reason, even though that isn't what abort() is for...

Will investigate...

**Comment by michaelrsweet** [comment]
Hmm, I'm having trouble reproducing this. Will continue to try...

**Comment by zdohnal** [comment]
Hi Mike,

Probably I'm able to reproduce this behavior with:

```$ lprint server -o server-hostname=localhost -o server-port=15000```

and then I send SIGINT via ctrl+c in terminal. I got the following output (I have lprint compiled with -DDEBUG):

```
subcommand='server'
num_options=2
options[0].name='server-hostname', value='localhost'
options[1].name='server-port', value='15000'
num_files=0
lprint: thread-watch.c:171: avahi_threaded_poll_lock: Assertion `p' failed.
Aborted (core dumped)
```

Here is the full output of ```coredumpctl gdb```  
[lprint-server.txt](https://github.com/michaelrsweet/lprint/files/5401731/lprint-server.txt)
.

The core of the issue (at least in my case) is that ```dnssd_master``` global variable is NULL, as you can see in the file.

I'll look into it tomorrow if I can propose a patch for fixing it (probably free only when dnssd_master isn't NULL).

**Comment by michaelrsweet** [comment]
It is possible this is something I've already fixed in PAPPL as well - the code using Avahi there is much more mature. I'm just about ready to start 1.0 betas for PAPPL and will refactor LPrint to use it instead.

**Comment by michaelrsweet** [comment]
Calling this fixed with the PAPPL 1.1 integration.
